superset/db_engine_specs/singlestore.py

```diff
@@ -0,0 +1,558 @@
+            "MOD",
+            "MONTH",
+            "MONTHNAME",
+            "MONTHS_BETWEEN",
+            "NOPARAM",
+            "NOW",
+            "NTH_VALUE",
+            "NTILE",
+            "NULLIF",
+            "NVL",
+            "IFNULL",
+            "PERCENT_RANK",
+            "PERCENTILE_CONT",
+            "MEDIAN",
+            "PERCENTILE_DISC",
+            "PI",
+            "POW",
+            "QUARTER",
+            "QUOTE",
+            "RADIANS",
+            "RAND",
+            "RANK",
+            "REDUCE",
+            "REGEXP_INSTR",
+            "REGEXP_MATCH",
+            "REGEXP_REPLACE",
+            "REGEXP_SUBSTR",
+            "REPLACE",
+            "REVERSE",
+            "RIGHT",
+            "RLIKE",
+            "REGEXP",
+            "ROUND",
+            "ROW_COUNT",
+            "ROW_NUMBER",
+            "RPAD",
+            "RTRIM",
+            "SCALAR_VECTOR_MUL",
+            "SEC_TO_TIME",
+            "SECOND",
+            "SECRET",
+            "SET",
+            "SHA1",
+            "SHA2",
+            "SIGMOID",
+            "SIGN",
+            "SIN",
+            "SLEEP",
+            "SPLIT",
+            "SQRT",
+            "STD",
+            "STDDEV",
+            "STDDEV_POP",
+            "STDDEV_SAMP",
+            "STR_TO_DATE",
+            "strcmp",
+            "STRING_BYTES",
+            "SUBSTRING",
+            "SUBSTRING_INDEX",
+            "SUM",
+            "SYS_GUID",
+            "UUID",
+            "TAN",
+            "TIME",
+            "TIME_BUCKET",
+            "TIME_FORMAT",
+            "TIME_TO_SEC",
+            "TIMEDIFF",
+            "TIMESTAMP",
+            "TIMESTAMPADD",
+            "TIMESTAMPDIFF",
+            "TO_BASE64",
+            "TO_CHAR",
+            "TO_DATE",
+            "TO_DAYS",
+            "TO_JSON",
+            "TO_NUMBER",
+            "TO_SECONDS",
+            "TO_TIMESTAMP",
+            "TRIM",
+            "TRUNC",
+            "TRUNCATE",
+            "UCASE",
+            "UNHEX",
+            "UNIX_TIMESTAMP",
+            "USER",
+            "UTC_DATE",
+            "UTC_TIME",
+            "UTC_TIMESTAMP",
+            "UUID_TO_BIN",
+            "VARIANCE",
+            "VAR_SAMP",
+            "VECTOR_ADD",
+            "VECTOR_ELEMENTS_SUM",
+            "VECTOR_KTH_ELEMENT",
+            "VECTOR_MUL",
+            "VECTOR_NUM_ELEMENTS",
+            "VECTOR_SORT",
+            "VECTOR_SUB",
+            "VECTOR_SUBVECTOR",
+            "VECTOR_SUM",
+            "WEEK",
+            "WEEKDAY",
+            "YEAR",
+        ]
+
+        if (database_name := cls.get_default_schema(database, None)) is not None:
+            df = database.get_df(
+                f"SHOW FUNCTIONS IN `{database_name.replace('`', '``')}`"
+            )
+
+            function_name_col = f"Functions_in_{database_name}"
+            if function_name_col in df:
+                functions += df[function_name_col].tolist()
```

Comment:
It's better to make `functions` a `set[str]`, then here you can:

```python
functions.update(df[function_name_col].tolist())  # .tolist() might not be necessary
```

Then return `sorted(functions)`. This way you remove duplicates.

Reply:
Also, we could access the dataframe by index, then we don't have to worry about the column name:

```python
functions.update(df.iloc[:, 0].tolist())
```

Reply:
Updated this